src/components/SourceCodeEditor.tsx

```diff
@@ -24,61 +25,65 @@ const SourceCodeEditor = (props: ISourceCodeEditor) => {
   const editorRef = useRef<Parameters<EditorDidMount>[0] | null>(null);
   const { sourceCode, setSourceCode } = useIntegrationSourceStore();
   const [syncedCode, setSyncedCode] = useState('');
-  const { integrationJson, updateIntegration } = useIntegrationJsonStore((state) => state);
   const { settings, setSettings } = useSettingsStore();
-  const previousJson = usePrevious(integrationJson);
+  const { flows, properties, setFlows } = useFlowsStore(({ flows, properties, setFlows }) => ({ flows, properties, setFlows }), shallow);
+  const previousFlows = usePrevious(flows);
+
   const schemaUri = settings.dsl.validationSchema
     ? process.env.KAOTO_API + settings.dsl.validationSchema
     : '';
 
   useEffect(() => {
-    let tmpInt: IIntegration = integrationJson;
-    if (previousJson === integrationJson) return;
+    if (previousFlows === flows || !Array.isArray(flows[0].steps)) return;
 
-    if (tmpInt.dsl != null && tmpInt.dsl !== settings.dsl.name) {
+    if (flows[0].dsl !== null && flows[0].dsl !== settings.dsl.name) {
       fetchCapabilities().then((capabilities: ICapabilities) => {
         capabilities.dsls.forEach((dsl) => {
-          if (dsl.name === tmpInt.dsl) {
+          if (dsl.name === flows[0].dsl) {
             const tmpSettings = { ...settings, dsl: dsl };
             setSettings(tmpSettings);
-            fetchTheSourceCode(tmpInt, tmpSettings);
+            fetchTheSourceCode({flows, properties}, tmpSettings);
           }
         });
       });
     } else {
-      fetchTheSourceCode(integrationJson, settings);
+      fetchTheSourceCode({flows, properties}, settings);
     }
-  }, [integrationJson]);
-
-  const fetchTheSourceCode = (integration: IIntegration, settings: ISettings) => {
-    const tmpIntegration = {
-      ...integration,
-      metadata: { ...integrationJson.metadata, ...settings },
-      dsl: settings.dsl.name,
+  }, [flows, properties]);
+
+  const fetchTheSourceCode = (currentFlowsWrapper: IFlowsWrapper, settings: ISettings) => {
+    const updatedFlowWrapper = {
```

Comment:
And this one is also a candidate in `FlowsService`, then can we put all YAML>JSON and JSON>YAML work in a single service?

Reply:
Yup, that makes sense :ok_hand: